lib/msal-core/dist/msal.js

```diff
@@ -2577,13 +2587,22 @@ var UserAgentApplication = /** @class */ (function () {
                 }
                 var scope_1 = scopes.join(" ").toLowerCase();
                 var userObject_1 = user ? user : _this.getUser();
-                if (!userObject_1) {
+                var adalIdToken = _this._cacheStorage.getItem(Constants_1.Constants.adalIdToken);
+                if (!userObject_1 && (extraQueryParameters && (extraQueryParameters.indexOf('login_hint') == -1 || extraQueryParameters.indexOf('sid') == -1)) && Utils_1.Utils.isEmpty(adalIdToken)) {
+                    _this._logger.info('User login is required');
                     reject(Constants_1.ErrorCodes.userLoginError + "|" + Constants_1.ErrorDescription.userLoginError);
                     return;
                 }
+                //if user didn't passes the login_hint and adal's idtoken is present and no userobject, user the login_hint gfrom adal's idToken
+                else if (!userObject_1 && !extraQueryParameters && !Utils_1.Utils.isEmpty(adalIdToken)) {
+                    var idTokenObject = Utils_1.Utils.extractIdToken(adalIdToken);
+                    console.log("ADAL's idToken exists. Extracting login information from ADAL's idToken ");
+                    extraQueryParameters = "&login_hint=" + idTokenObject.upn;
+                }
                 var authenticationRequest_1;
                 var newAuthority = authority ? AuthorityFactory_1.AuthorityFactory.CreateInstance(authority, _this.validateAuthority) : _this.authorityInstance;
-                if (Utils_1.Utils.compareObjects(userObject_1, _this.getUser())) {
+                var msalIdToken = _this._cacheStorage.getItem(Constants_1.Constants.idTokenKey);
+                if (Utils_1.Utils.compareObjects(userObject_1, _this.getUser()) && !Utils_1.Utils.isEmpty(msalIdToken)) { //TODO revisit this change if getUser() is not null
```

Comment:
It will go to the else block and get both idToken and accesstoken. Notice that the responsetype is idtoken and token
else {
          authenticationRequest = new AuthenticationRequestParameters(newAuthority, this.clientId, scopes, ResponseTypes.id_token_token, this._redirectUri, this._state);
        }